HARK/distribution.py

```diff
@@ -1192,6 +1192,10 @@ def RNG(self):
         """
         return self.dataset.RNG
 
+    @RNG.setter
+    def RNG(self,value):
+        self.dataset.attrs['RNG'] = value
```

Comment:
Should this be `np.random.RandomState(value)`?

Reply:
I am not sure? What does this property do?

I wrote as it is because I thought that the `get` and `set` method should deal with exactly the same object. Since the getter is
https://github.com/econ-ark/HARK/blob/daf2afd9243bf83a9bbd3477c3f14b09b1d52538/HARK/distribution.py#L1188-L1193

I wrote the setter to the RNG property directly. What do you think?

Reply:
I think this is how we initialize a random state with a seed. When we initialize a DiscreteDsitribution object, we set RNG as follows 

https://github.com/econ-ark/HARK/blob/daf2afd9243bf83a9bbd3477c3f14b09b1d52538/HARK/distribution.py#L1079

so I guess we would have to do `dist.RNG = np.random.RandomState(seed)`. 

Actually I guess this isn't a problem, you just have to be more explicit that the `value` parameter should be of type `RandomState`. Maybe just a type check?

But also if we have a setter for `dist.seed = value` it should also generate a new RNG.

Reply:
I like the typecheck solution.

You did not add a setter for the seed property (I think?) Should I create one? Would that go in a different PR?

Reply:
Alrighty, added  a typecheck.

I could not remember why I had to add this setter in the first place, but just ran some tests and found out. The reason was that methods like `initialize_sim` call `Distribution.reset()` on things like the income distribution. That method is
https://github.com/econ-ark/HARK/blob/20645e2bae21e3af8bd06d3dcde99ea4b00dc8fe/HARK/distribution.py#L24-L31

So without a setter, it failed